src/software/backend/input/network/network_filter.cpp

```diff
@@ -0,0 +1,392 @@
+#include "software/backend/input/network/network_filter.h"
+
+#include "shared/constants.h"
+#include "software/constants.h"
+#include "software/parameter/dynamic_parameters.h"
+#include "software/proto/messages_robocup_ssl_detection.pb.h"
+#include "software/proto/messages_robocup_ssl_geometry.pb.h"
+#include "software/sensor_fusion/refbox_data.h"
+
+// We can initialize the field_state with all zeroes here because this state will never
+// be accessed by an external observer to this class. the getFieldData must be called to
+// get any field data which will update the state with the given protobuf data
+NetworkFilter::NetworkFilter()
```

Comment:
This _really_ doesn't feel like a filter class at all. Perhaps "SSLProtobufReader" might be more appropriate? @MathewMacDougall ?

Reply:
I agree this is not longer a filter class. `SSLProtobufReader` sounds reasonable.

Reply:
changed